### PR TITLE
DVCSMP-2391 Iris 4 button device mapping to button 1 only

### PR DESCRIPTION
**Pull request by workingmonk**
cc @tpmanley

### CODE REVIEW

File: devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy [hunk 1]
```diff
@@ -118,15 +118,43 @@ private Map getBatteryResult(rawValue) {
...
         }
         else if (descMap.command == "00") {
             getButtonResult("release")
         }
     }
+    else if ((device.getDataValue("model") == "3450-L") && (descMap.clusterInt == 0x0006)) {
+    	log.info "For 3450-L"
+        log.trace "descMap : $descMap"
+        if (descMap.command == "01") {
+            getButtonResult("press")
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy]: Do you need the button number here too?

File: devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy [hunk 2]
```diff
@@ -118,15 +118,43 @@ private Map getBatteryResult(rawValue) {
...
+        }
+        else if (descMap.command == "00") {
+        	def button = 1
+            switch(descMap.sourceEndpoint) {
+            	case "01":
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy]: Does this mapping correspond to left to right, top to bottom?

File: devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy [hunk 3]
```diff
@@ -118,15 +118,43 @@ private Map getBatteryResult(rawValue) {
...
             getButtonResult("press")
         }
         else if (descMap.command == "00") {
             getButtonResult("release")
         }
     }
+    else if ((device.getDataValue("model") == "3450-L") && (descMap.clusterInt == 0x0006)) {
+    	log.info "For 3450-L"
+        log.trace "descMap : $descMap"
+        if (descMap.command == "01") {
```
Comment (varzac) [inline devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy]: `commandInt` is probably better to match on.

File: devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy [hunk 4]
```diff
@@ -118,15 +118,43 @@ private Map getBatteryResult(rawValue) {
...
+            getButtonResult("press")
+        }
+        else if (descMap.command == "00") {
+        	def button = 1
+            switch(descMap.sourceEndpoint) {
+            	case "01":
+                	button = 4
+                    break
+                case "02":
```
Comment (varzac) [inline devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy]: Are these really the correct endpoint->button mappings... weird

**Review by varzac — APPROVED** [review]

File: devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy [hunk 5]
```diff
@@ -118,15 +118,43 @@ private Map getBatteryResult(rawValue) {
...
+                	button = 3
+                    break
+                case "03":
+                	button = 1
+                    break
+                case "04":
+                	button = 2
+                    break
+                default:
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy]: Don't need the default case since it is initialized to 1

**Review by tpmanley — COMMENTED** [review]
Minor, but the indentation is inconsistent

File: devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy (hunk 4 above)
Comment (workingmonk) [reply devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy]: it is based on how we number the buttons on other devices

File: devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy (hunk 5 above)
Comment (workingmonk) [reply devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy]: Yes

File: devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy (hunk 2 above)
Comment (workingmonk) [reply devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy]: Correct

File: devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy (hunk 1 above)
Comment (workingmonk) [reply devicetypes/smartthings/zigbee-button.src/zigbee-button.groovy]: Not really. the event is generated only with release

**Comment by workingmonk** [comment]
@tpmanley @varzac incorporated suggestions from both of you

**Comment by varzac** [comment]
:+1:

**Review by tpmanley — APPROVED** [review]

**Pull request merged**